### DISCUSSION TITLE
SequelizeAccessDeniedError: Access denied for user 'root'@'localhost' (using password: YES)

### DISCUSSION THREAD

**Discussion by JavierPons** [sequelize/sequelize#13020] [Help & Questions] [answered]
Hi all, after been searching for all day I decided to ask here if someone can give some lights to this issue. Why and how to solve this.
with homebrew I have installed mysql and it is running.

macOS Catalina 10.15.6
Node v10.15.3
Sequelize v6.5.0
mysql2 v2.2.5

app.js  running the server and added

`// DB
const db = require('./models/index')
async ()=> {
    try{
        await db.sync()
    }catch(e){
        console.log(e);
    }
};
`

config.js

`module.exports = {
    DB_HOST: "127.0.0.1",
    DB_USER: "root",
    DB_PASSWORD: "123456",
    DB_NAME: "farmerdb",
    dialect : "mysql",
    PORT: 3306 
}`

models/model.js

`const { Sequelize } = require("sequelize");
const db = require("./index")

    const Farmer = db.define("farmerdb",{
        farmID:{
            type: Sequelize.INTEGER,
            primaryKey: true,
            autoIncrement: true
        },
        name: {
            type: Sequelize.STRING
        },
        tel: {
            type: Sequelize.STRING
        },
        gender: {
            type: Sequelize.STRING
        }
    });
   
module.exports = Farmer;`

models/index.js

`const dbConfig = require('../config/config');
const Sequelize = require('sequelize');

const sequelize = new Sequelize(dbConfig.DB_NAME, dbConfig.DB_USER, dbConfig.DB_PASSWORD,{
    host: dbConfig.DB_HOST,
    dialect: "mysql",
    port: dbConfig.PORT,
    logging: false
});

sequelize.authenticate().then(()=> {
    console.log("Connection has been stablised successfully.");
}).catch(err => {
    console.log("Unable to connect to the database" , err);
})

module.exports = sequelize;
`

**Comment by JavierPons** [comment, marked as answer]
Fixed.  The password was incorrect.

**Reply by aham-uzoma** [reply]
what password exactly, do you mean the exact root password for the database?

**Comment by geekcoldhand** [comment]
you need to make sure you have sourced your db schema before you run the server in Sequelize. Because Sequelize looks for the data

**Comment by ErickDev2** [comment]
Tengo este error directamente en la terminal de nodemon, ya intente con varias cosas pero me sigue saliendo el mismo error.

Alguien que me pueda ayudar

sqlMessage: "Access denied for user 'root'@'localhost' (using password: YES)",

**Comment by St80ene** [comment]
In my case, I was using the wrong username. I hope this information is helpful to somebody.
